langmeta-sqlite/src/main/resources/semanticdb.ddl

```diff
@@ -0,0 +1,49 @@
+create table document(
+  id int,
+  filename text,
```

Comment:
In general, how open would you be to refactoring the protobuf schema? There are some things that we could do to it in order to ensure parity with the SQL schema:
  * Fix small inconsistencies (e.g. `Synthetic.pos` vs everyone else's `position` field).
  * Inline positions to save on keystrokes.
  * Split `Denotation.flags` into bits.

Looks like all these changes can be performed in a backward-compatible manner, so that existing programs written against the current protobuf schema continue working with the schema. Source compatibility of the lowlevel layer can be ensured via extension methods (or can be forgone completely since the lowlevel layer contains `internal` in its package name).

Reply:
/cc @olafurpg

Reply:
Why split denotation flags into bits?

Reply:
To make it easy to query for individual bits. `(s.flags & 135168) = 135168` looks awful to a newcomer. `s.is_class && s.is_final` is much better`. Same about SQL.

Reply:
Let’s test this.  It may be a large performance hit.  In general sql engines don’t consider Boolean flags for indexing, because they don’t offer enough discrimination to offset the lookup cost to the index, (there are two total values by definition) so besides the space usage of the fact that SQLite has no true Boolean type and will store these as an int under the covers, queries would get slower.  

It would be better I think to add a function as @olafurpg mentioned, to the client library if we are offering that for re-use or to the db itself if we expect people to connect directly to the db and ignore our library when analyzing it.